### PR TITLE
Add PUT and POST Endpoints for SSP Implemented Requirement

### PR DESCRIPTION
**Pull request by hreineck**
Add documentation for a new endpoint for a PUT request that updates
an implemented requirement in an SSP.

Closes EasyDynamics/easygrc#19

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -1844,6 +1844,68 @@ paths:
...
+          description: Invalid ID supplied
+          content: {}
+        404:
+          description: System security plan not found
+          content: {}
+      security:
+        - oscal_auth:
+            - write:ssps
+            - read:ssps
+  /system-security-plans/{sspId}/control-implementation/implemented-requirements/{implementedRequirementID}:
```
Comment (pkothare) [inline openapi.yaml]:
```suggestion
  /system-security-plans/{sspId}/control-implementation/implemented-requirements/{implementedRequirementId}:
```
.. Just to keep naming consistent.

File: openapi.yaml [hunk 2]
```diff
@@ -1844,6 +1844,68 @@ paths:
...
+          in: header
+          schema:
+            type: string
+        - name: sspId
+          in: path
+          description: System security plan ID
+          required: true
+          schema:
+            type: string
+        - name: implementedRequirementID
```
Comment (pkothare) [inline openapi.yaml]:
```suggestion
        - name: implementedRequirementId
```

File: openapi.yaml [hunk 3]
```diff
@@ -1844,6 +1844,68 @@ paths:
...
             - write:ssps
             - read:ssps
+  /system-security-plans/{sspId}/control-implementation/implemented-requirements:
+    post:
+      tags:
+        - OSCAL System Security Plan
+      summary: Add an implemented requirement to an OSCAL system security plan
+      operationId: addImplementedRequirementToSsp
+      parameters:
+        - name: api_key
```
Comment (pkothare) [inline openapi.yaml]: Are we including `api_key` on all operations as part of the spec? Maybe @rgauss you can give me a quick yes/no on this.

**Review by pkothare — CHANGES_REQUESTED** [review]
Requesting minor tweaks based on the naming convention we've used.

File: openapi.yaml (hunk 3 above)
Comment (rgauss) [reply openapi.yaml]: This is really a holdover example.  This and things like the `oscal_auth` references need to be revamped at some point.

In short, no, we don't need it.

**Review by rgauss — CHANGES_REQUESTED** [review]
We'll need to specify what `requestBody` looks like on these as well.

See the [OSCALParty schema](https://github.com/EasyDynamics/oscal-rest/blob/develop/openapi.yaml#L2097) for precedence on how to pick out a 'subobject' from a root OSCAL object.

File: openapi.yaml [hunk 4]
```diff
@@ -1844,6 +1844,82 @@ paths:
...
+          in: header
+          schema:
+            type: string
+        - name: sspId
+          in: path
+          description: System security plan ID
+          required: true
+          schema:
+            type: string
+      responses:
```
Comment (rgauss) [inline openapi.yaml:1867]: Should we add a `201` response here?

File: openapi.yaml (hunk 3 above)
Comment (rgauss) [reply openapi.yaml]: @hreineck, can we remove these?

File: openapi.yaml [hunk 5]
```diff
@@ -1844,6 +1844,82 @@ paths:
...
+          required: true
+          schema:
+            type: string
+        - name: implementedRequirementId
+          in: path
+          description: ID of implemented requirement to be updated
+          required: true
+          schema:
+            type: string
+      responses:
```
Comment (rgauss) [inline openapi.yaml:1912]: Should we add a `200` here?

File: openapi.yaml [hunk 6]
```diff
@@ -2088,6 +2164,10 @@ components:
       properties:
         system-security-plan:
           $ref: "#/components/schemas/OSCALPartialUpdateExample"
+    OSCALImplementedRequirement:
+      type: object
+      $ref:
+        "https://raw.githubusercontent.com/usnistgov/OSCAL/main/json/schema/oscal_ssp_schema.json#assembly_oscal-ssp_implemented-requirement"
```
Comment (rgauss) [inline openapi.yaml]: This currently needs to reference the `EasyDynamics` fork like the other schema refs (though NIST may be fixing that soon).

You can double check by trying to preview this file in the online Swagger editor.

**Review by rgauss — CHANGES_REQUESTED** [review]

**Review by mikeisen1 — CHANGES_REQUESTED** [review]
Everything looks good except we need a valid reference to the implemented-requirements schema.

File: openapi.yaml [hunk 7]
```diff
@@ -2088,6 +2170,10 @@ components:
       properties:
         system-security-plan:
           $ref: "#/components/schemas/OSCALPartialUpdateExample"
+    OSCALImplementedRequirement:
+      type: object
+      $ref:
+        "https://raw.githubusercontent.com/EasyDynamics/OSCAL/main/json/schema/oscal_ssp_schema.json#assembly_oscal-ssp_implemented-requirement"
```
Comment (mikeisen1) [inline openapi.yaml]:
```suggestion
    OSCALImplementedRequirement:
      type: object
      properties:
        implemented-requirement:
          $ref:
            "https://raw.githubusercontent.com/EasyDynamics/OSCAL/json-schema-ref-by-path/json/schema/oscal_ssp_schema.json#/definitions/assembly_oscal-ssp_implemented-requirement"
```

**Review by mikeisen1 — APPROVED** [review]

File: openapi.yaml [hunk 8]
```diff
@@ -1844,6 +1844,88 @@ paths:
...
+          content: {}
+      security:
+        - oscal_auth:
+            - write:ssps
+            - read:ssps
+  /system-security-plans/{sspId}/control-implementation/implemented-requirements/{implementedRequirementId}:
+    put:
+      tags:
+        - OSCAL System Security Plan
+      summary: Updates the body of an implemented requirement in an OSCAL system security plan
```
Comment (rgauss) [inline openapi.yaml]:
```suggestion
      summary: Updates an implemented requirement in an OSCAL system security plan
```

**Review by rgauss — APPROVED** [review]
Minor suggestion on description that does not have to hold up review.

**Comment by rgauss** [comment]
@pkothare, does this look good to you now?

**Pull request merged**